types/index.d.ts

```diff
@@ -0,0 +1,507 @@
+// Minimum TypeScript Version: 3.8
+declare module '@okta/okta-signin-widget';
+import { OktaAuth, OktaAuthOptions, Tokens } from '@okta/okta-auth-js';
+/* eslint-disable camelcase */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+export default class OktaSignIn implements OktaSignIn {
+  constructor(config: OktaSIW.WidgetConfig);
+
+  authClient: OktaAuth;
+
+  on(event: 'ready', callback: OktaSIW.EventCallback): void;
+  on(event: 'afterError', callback: OktaSIW.EventCallbackWithError): void;
+  on(event: 'afterRender', callback: OktaSIW.EventCallback): void;
+
+  off(event?: 'ready', callback?: OktaSIW.EventCallback): void;
+  off(event?: 'afterError', callback?: OktaSIW.EventCallbackWithError): void;
+  off(event?: 'afterRender', callback?: OktaSIW.EventCallback): void;
+
+  show(): void;
+  hide(): void;
+  remove(): void;
+
+  showSignInToGetTokens(options: OktaSIW.ShowSignInToGetTokensOptions): Promise<Tokens>;
+  showSignInAndRedirect(options: OktaSIW.ShowSignInAndRedirectOptions): Promise<void>;
+  renderEl(
+    options: OktaSIW.RenderElOptions,
+    success?: (res: OktaSIW.RenderResult) => void,
+    error?: (err: OktaSIW.RenderError) => void
+  ): Promise<OktaSIW.RenderResult>;
+}
+
+export namespace OktaSIW {
```

Comment:
Why the creation of this namespace? Was there a problem with name collision?

Reply:
Renamed to `OktaSignIn`